xbmc/games/addons/GameClient.h

```diff
@@ -36,6 +36,31 @@ class CGameClientInput;
 class CGameClientProperties;
 class IGameInputCallback;
 
+/*!
+ * \ingroup games
+ * \brief Helper class to have "C" struct created before other parts becomes his pointer.
+ */
+class CGameClientStruct
```

Comment:
This class is added to create the "C" API structure before any other parts becomes started.

Seen during runtime test, that without **`CGameClientInput`** produce a crash as his copied **`AddonProps_Game`** pointer was `NULL`.

It works also to give him the complete structure **`AddonInstance_Game`**, but as  **`CGameClientSubsystem`** is contructed before the memory of "C" is created, can never be for future sure that something becomes added where access the "C" part before.